pkg/apis/build/v1alpha1/build_validation.go

```diff
@@ -68,15 +73,15 @@ func (b *TemplateInstantiationSpec) Validate() *apis.FieldError {
 		return nil
 	}
 	if b.Name == "" {
-		return apis.ErrMissingField("build.spec.template.name")
+		return apis.ErrMissingField("name").ViaField("template")
```

Comment:
The `.ViaField` should be in the caller (we don't know what field this was accessed through.

Reply:
I can see how the best pattern to follow for `ViaField()` is to add it to the caller, mainly because of the reason you put, but for the sake of learning: could `Validate` implemented as part of an TemplateInstantiationSpec interface ever not be called by the `template` field?

That being asked, I will definitely go ahead and move the `ViaField()` to the caller.